tests/kafkatest/services/kafka/kafka.py

```diff
@@ -165,6 +165,24 @@ def start(self, add_principals=""):
         self.start_minikdc(add_principals)
         Service.start(self)
 
+        self.logger.info("Waiting for brokers to register at ZK")
+
+        retries = 30
+        expected_broker_ids = set(self.nodes)
+        broker_ids = set()
+        while retries > 0:
+            for node in self.nodes:
+                if self.is_registered(node):
```

Comment:
This computation of `broker_ids` can be simpler with a set comprehension like `{node for node in self.nodes if self.is_registered(node)}`

Reply:
Don't speak Python... Without set comprehension using `wait_until` was not really possible...

Reply:
no worries. fyi, you still can, you just define a local function that has your check logic. you can `def foo()` anywhere.